app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+  def login
+    user = User.find_by(email: params[:email])
+    if user&.authenticate(params[:password])
+      render json: { message: 'Logged in successfully!', token: user.auth_token }
+    else
+      render json: { error: 'Invalid email or password' }, status: :unauthorized
+    end
+  end
+
+  def logout
+    auth_token = request.headers['Authorization']
+    user = User.find_by(auth_token: auth_token)
+    if user
+      user.update(auth_token: nil)
```

Comment:
This is not what logout means.  @neeraj: Share spec of logout feature, else remove this feature for now. It makes more sense for sessions, or places where we have multiple tokens out of which we can expire one